plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/Activator.java

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.appengine.localserver;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+import org.osgi.framework.BundleContext;
+
+/**
+ * Activator class that controls the plugin lifecycle.
+ */
+public class Activator extends AbstractUIPlugin {
+  // The plug-in ID
+  public static final String PLUGIN_ID = "com.google.cloud.tools.eclipse.appengine.localserver";
+
+  // The shared instance
+  private static Activator plugin;
+
+  /**
+   * Returns the shared instance.
+   */
+  public static Activator getDefault() {
+    return plugin;
+  }
+
+  @Override
+  public void start(BundleContext context) throws Exception {
+    super.start(context);
+    plugin = this;
+  }
+
+  @Override
+  public void stop(BundleContext context) throws Exception {
+    plugin = null;
+    super.stop(context);
+  }
+
+  /**
+   * Logs the specified message and opens an error dialog with the specified
+   * message and title. Ensures that the error dialog is opened on the UI
+   * thread.
+   *
+   * @param parent the parent shell of the dialog, or null if none
+   * @param title the dialog's title
+   * @param message the message
+   */
+  public static void logAndDisplayError(final Shell parent,
```

Comment:
The better approach is to use the JFace `org.eclipse.ui.statushandler.StatusHandlers` class